release/serve_tests/workloads/scalability_1k_replicas.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import click
+import json
+import logging
+from typing import Optional
+
+from anyscale import service
+from anyscale.compute_config.models import (
+    ComputeConfig,
+    HeadNodeConfig,
+    WorkerNodeGroupConfig,
+)
+import ray
+
+from anyscale_service_utils import start_service
+from locust_utils import LocustLoadTestConfig, LocustStage, run_locust_load_test
+from serve_test_utils import save_test_results
+
+
+logger = logging.getLogger(__file__)
+
+DEFAULT_FULL_TEST_NUM_REPLICA = 1000
+DEFAULT_FULL_TEST_TRIAL_LENGTH_S = 60
+
+
+@click.command()
+@click.option("--num-replicas", type=int)
+@click.option("--trial-length", type=str)
+def main(num_replicas: Optional[int], trial_length: Optional[str]):
+    num_replicas = num_replicas or DEFAULT_FULL_TEST_NUM_REPLICA
+    trial_length = trial_length or DEFAULT_FULL_TEST_TRIAL_LENGTH_S
+    noop_1k_application = {
+        "name": "default",
+        "import_path": "noop:app",
+        "route_prefix": "/",
+        "runtime_env": {"working_dir": "workloads"},
+        "deployments": [
+            {
+                "name": "Noop",
+                "num_replicas": num_replicas,
+                "ray_actor_options": {"resources": {"worker_resource": 0.01}},
+            }
+        ],
+    }
+    compute_config = ComputeConfig(
+        cloud="anyscale_v2_default_cloud",
+        head_node=HeadNodeConfig(instance_type="m5.8xlarge"),
+        worker_nodes=[
+            WorkerNodeGroupConfig(
+                instance_type="m5.8xlarge",
+                min_nodes=0,
+                max_nodes=50,
```

Comment:
I think we probably want to run many more nodes than this. in most realistic applications users are running ~1-8 replicas per node

Reply:
let's set a max of 4 replicas per node